sogs/model/room.py

```diff
@@ -259,6 +276,32 @@ def image(self, file: Union[File, int]):
 
             self._refresh()
 
+    @property
+    def pinned_messages(self):
+        """
+        Accesses the list of pinned messages for this room; this is fetched from the database the
+        first time this is accessed.  Each element is a dict such as:
+        {"id": 1234, "pinned_at": 1642701309.5007384, "pinned_by": "05123456....."}
+        The returned list should not be modified; instead call pin() or unpin() to add/remove pinned
+        messages.
+        """
+
+        if self._pinned is None:
+            self._pinned = [
+                {'id': r[0], 'pinned_at': r[1], 'pinned_by': r[2]}
+                for r in query(
+                    """
+                    SELECT message, pinned_at, users.session_id
+                    FROM pinned_messages JOIN users ON pinned_by = users.id
+                    WHERE room = :r
+                    ORDER BY pinned_at
+                    """,
+                    r=self.id,
+                )
+            ]
+
+        return self._pinned
```

Comment:
we could do 
`return list(self._pinned)`
so we dont need to tell the user it's supposed to be immutable